src/renderer/components/+extensions/installed-extensions.tsx

```diff
@@ -46,62 +46,58 @@ function getStatus(extension: InstalledExtension) {
-  const columns = useMemo(
-    () => [
-      {
-        Header: "Name",
-        accessor: "extension",
-        width: 200,
-        sortType: (rowA: Row, rowB: Row) => { // Custom sorting for extension name
-          const nameA = extensions[rowA.index].manifest.name;
-          const nameB = extensions[rowB.index].manifest.name;
-
-          if (nameA > nameB) return -1;
-          if (nameB > nameA) return 1;
-
-          return 0;
-        },
-      },
-      {
-        Header: "Version",
-        accessor: "version",
-      },
-      {
-        Header: "Status",
-        accessor: "status",
-      },
-      {
-        Header: "",
-        accessor: "actions",
-        disableSortBy: true,
-        width: 20,
-        className: "actions",
+  const cols = useMemo<ColumnDef<InstalledExtension>[]>(() => ([
+    {
+      id: "name",
+      header: "Name",
+      size: 200,
+      sortingFn: (a, b) => { // Custom sorting for extension name
+        const nameA = extensions[a.index].manifest.name;
+        const nameB = extensions[b.index].manifest.name;
+
+        if (nameA > nameB) return -1;
+        if (nameB > nameA) return 1;
+
+        return 0;
       },
-    ], [],
-  );
-
-  const data = useMemo(
-    () => extensions.map(extension => {
-      const { id, isEnabled, isCompatible, manifest } = extension;
-      const { name, description, version } = manifest;
-      const isUninstalling = extensionInstallationStateStore.isExtensionUninstalling(id);
-
-      return {
-        extension: (
-          <div className={"flex items-start"}>
-            <div>
-              <div className={styles.extensionName}>{name}</div>
-              <div className={styles.extensionDescription}>{description}</div>
-            </div>
+      cell: (cell) => (
+        <div className={"flex items-start"}>
+          <div>
+            <div className={styles.extensionName}>{cell.row.original.manifest.name}</div>
+            <div className={styles.extensionDescription}></div>
           </div>
-        ),
-        version,
-        status: (
+        </div>
+      )
+    },
+    {
+      id: "version",
+      header: "Version",
+      cell: (cell) => cell.row.original.manifest.version
+    },
+    {
+      id: "status",
+      header: "Status",
+      accessorFn: (row) => row,
+      cell: (cell) => {
+        const extension = cell.row.original;
+        const { isEnabled, isCompatible } = extension;
+        
+        return (
           <div className={cssNames({ [styles.enabled]: isEnabled, [styles.invalid]: !isCompatible })}>
             {getStatus(extension)}
           </div>
-        ),
-        actions: (
+        )
+      }
+    },
+    {
+      id: "actions",
+      enableSorting: false,
+      size: 30,
+      cell: (cell) => {
+        const extension = cell.row.original;
+        const { id, isEnabled, isCompatible } = extension;
+        const isUninstalling = extensionInstallationStateStore.isExtensionUninstalling(id);
```

Comment:
Logic about "isUninstalling" does not quite belong here in UI-code, but instead on the model of the row, preferably as `IComputedValue`.